package/MDAnalysis/converters/OpenMMParser.py

```diff
@@ -102,42 +105,81 @@ def _mda_topology_from_omm_topology(self, omm_topology):
         -------
         top : MDAnalysis.core.topology.Topology
 
-        """
 
+        .. versionchanged:: 2.1.0
+           * The parser now works when element attribute is missing in some or
+             all the atoms.
+           * When none of the elements are present their atomtypes are guessed using
+             their names and their masses are then guessed using their atomtypes.
+           * When partial elements are present, values from available elements are used
+             whereas the absent elements are assigned an empty string with their
+             atomtype set to 'X' and mass set to 0.0.
+
+        """
         atom_resindex = [a.residue.index for a in omm_topology.atoms()]
         residue_segindex = [r.chain.index for r in omm_topology.residues()]
         atomids = [a.id for a in omm_topology.atoms()]
         atomnames = [a.name for a in omm_topology.atoms()]
         chainids = [a.residue.chain.id for a in omm_topology.atoms()]
-        elements = [a.element.symbol for a in omm_topology.atoms()]
-        atomtypes = [a.element.symbol for a in omm_topology.atoms()]
-        masses = [a.element.mass._value for a in omm_topology.atoms()]
         resnames = [r.name for r in omm_topology.residues()]
         resids = [r.index + 1 for r in omm_topology.residues()]
         resnums = resids.copy()
         segids = [str(c.index) for c in omm_topology.chains()]
         bonds = [(b.atom1.index, b.atom2.index) for b in omm_topology.bonds()]
         bond_orders = [b.order for b in omm_topology.bonds()]
         bond_types = [b.type for b in omm_topology.bonds()]
-
-        n_atoms = len(atomids)
-        n_residues = len(resids)
-        n_segments = len(segids)
+        elements = [None if a.element is None else a.element.symbol
+                    for a in omm_topology.atoms()]
```

Comment:
You don't really use the contents of this list. Could you reorder this to:

* construct a validated_elements list and masses at the same time
* Check the values and give the appropriate warning if needed
* Guess/construct the atomtypes?

Reply:
The list is actually being used to test whether all the elements are set or not, if some of the elements are present the code is pretty much already doing the things you requested. Do you want me to make further modifications please let me know

Reply:
> The list is actually being used to test whether all the elements are set or not

Yeah, I just think that this list does not need to be constructed and you could work the check into the remainder of the code. I suggested an outline above. It's not that the code is *wrong*, just that redundant code always adds maintenance burden.